.github/workflows/tests.yaml

```diff
@@ -48,15 +48,74 @@ jobs:
           name: vault-csi-provider-image
           path: ${{ env.TARBALL_FILE }}
 
+  vault-versions:
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo "setting vault versions"
+    outputs:
+      VAULT_N_2: 1.13.13
+      VAULT_N_1: 1.14.10
+      VAULT_N: 1.15.6
+
+  k8s-versions:
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo "setting k8s versions"
+    outputs:
+      K8S_1_25: 1.25.16
+      K8S_1_26: 1.26.14
+      K8S_1_27: 1.27.11
+      K8S_1_28: 1.28.7
+      K8S_1_29: 1.29.2
+
   integration-test:
+    runs-on: ubuntu-latest
     needs:
       - lint
       - test
+      - vault-versions
+      - k8s-versions
     strategy:
       fail-fast: false
       matrix:
-        kind-k8s-version: [1.25.16, 1.26.14, 1.27.11, 1.28.7, 1.29.2]
-    runs-on: ubuntu-latest
+        # test matrix setup is based off of https://github.com/hashicorp/vault-secrets-operator/blob/716db60e20c7873774374e6908c535e3f5f3e31a/.github/workflows/build.yaml#L261
+        kind-k8s-version:
+          - "${{ needs.k8s-versions.outputs.K8S_1_25 }}"
+          - "${{ needs.k8s-versions.outputs.K8S_1_26 }}"
+          - "${{ needs.k8s-versions.outputs.K8S_1_27 }}"
+          - "${{ needs.k8s-versions.outputs.K8S_1_28 }}"
+          - "${{ needs.k8s-versions.outputs.K8S_1_29 }}"
+        vault-version:
+          - "${{ needs.vault-versions.outputs.VAULT_N_2 }}"
+          - "${{ needs.vault-versions.outputs.VAULT_N_1 }}"
+          - "${{ needs.vault-versions.outputs.VAULT_N }}"
+        # Note: We want to test the operator with the different k8s versions,
+        # and with the different vault versions, but we don't care about testing
+        # all the k8s versions against all the vault versions.
+        # Combos to exclude:
+        #   kind-k8s-version: [K8S_1_25, K8S_1_26, K8S_1_27, K8S_1_28]
+        #   vault-version: [VAULT_N_2, VAULT_N_1]
+        exclude:
+          # vault version N-2
+          - kind-k8s-version: ${{ needs.k8s-versions.outputs.K8S_1_28 }}
```

Comment:
Similarly to above, I am not too opinionated on this, but one improvement we could make would be to extract the testing steps below into a reusable action and call it with 2 different matrices. In one, we would specify the latest Vault version and all k8s versions, and in the other we would specify the latest k8s version and all Vault versions. Then we could get rid of this repetitive/mechanical list of exclusions.